security/pkg/nodeagent/caclient/credentials.go

```diff
@@ -77,8 +69,8 @@ func (t *TokenProvider) RequireTransportSecurity() bool {
 // volatile memory), we can still proceed and allow other authentication methods to potentially
 // handle the request, such as mTLS.
 func (t *TokenProvider) GetToken() (string, error) {
-	if !t.forCA {
-		// When not using a CA, we only support reading from file
+	if !t.opts.SdsEnableTokenExchange && !t.opts.XdsEnableTokenExchange {
```

Comment:
I don't understand why there are split options (XDS and SDS), but we only ever check them together?

Reply:
What is your suggested revision?